### DISCUSSION TITLE
Re-exporting `http` from axum-core

### DISCUSSION THREAD

**Discussion by kellytk** [tokio-rs/axum#654] [Ideas]
I've a custom extractor project that uses `HeaderValue` and `StatusCode` from `http` with `axum-core`.  As that is already a dependency of `axum-core`, please consider re-exporting it similarly to [body](https://github.com/tokio-rs/axum/blob/main/axum-core/src/lib.rs#L58).

**Comment by davidpdrsn** [comment]
Can't you just depend on `http` directly? You don't _need_ the re-export and since using axum-core is rarer (just for libs) I'm not sure a re-export is necessary.

I get that it's a decent convenience but `cargo add http` is all it takes to fix it.

**Reply by kellytk** [reply]
Sure, that's how I have it working at present and it's perfectly sufficient.  Thank you.

If I may ask, in general, how do you determine when a lib should re-export a dependency?

**Reply by davidpdrsn** [reply]
> If I may ask, in general, how do you determine when a lib should re-export a dependency?

Honestly don't know 🤷‍♂️ I'm personally not a fan of re-exports at all. I find it makes dependencies less clear. It also leads to questions like this https://github.com/tokio-rs/axum/issues/651

However users _really_ wanted re-exports for axum, so we added some.

Adding this to axum-core _probably_ would be fine but I would like to be conservative and keep core as small as possible.

**Reply by kellytk** [reply]
You make a good argument for eschewing re-exports entirely.  Thanks once again.

**Reply by davidpdrsn** [reply]
Thanks for understanding 😊 who knows, maybe we'll reconsider if more people ask about this.
